src/interchange/src/avro/schema.rs

```diff
@@ -327,14 +327,20 @@ impl ConfluentAvroResolver {
                      that can lead to this branch"
                 );
                 // XXX(guswynn): use destructuring assignments when they are stable
-                let (schema_id, adjusted_bytes) = crate::confluent::extract_avro_header(bytes)?;
+                let (schema_id, adjusted_bytes) = match crate::confluent::extract_avro_header(bytes)
+                {
+                    Ok(ok) => ok,
+                    Err(err) => return Ok(Err(err)),
+                };
                 bytes = adjusted_bytes;
-                let schema = cache
+                let result = cache
                     .get(schema_id, &self.reader_schema)
-                    .await
-                    .with_context(|| {
-                        format!("failed to resolve Avro schema (id = {})", schema_id)
-                    })?;
+                    .await?
```

Comment:
this is the only transient error right? each place like this should be documented (as well as each function with a nested result)

Reply:
done